src/ralph/backends/http/async_lrs.py

```diff
@@ -174,13 +175,14 @@ def fetch_statements(params: dict):
                     params.update(parse_qs(urlparse(statements_response.more).query))
 
         try:
-            yield from fetch_statements(params=query_params)
+            async for statement in fetch_statements(params=query_params):
+                yield statement 
         except HTTPError as error:
             msg = "Failed to fetch statements."
             logger.error("%s. %s", msg, error)
             raise BackendException(msg, *error.args) from error
```

Comment:
The client connection will be finally closed here:

```suggestion
            raise BackendException(msg, *error.args) from error
        finally:
            client.close()
```

Reply:
Unless I misunderstood your comment, calling `client.close()` is not necessary as this is handled by the context manager:

`async with AsyncClient(auth=self.auth, headers=self.headers) as client:`

Reply:
Closing for now